src/library-authoring/component-info/ComponentAdvancedInfo.tsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable import/prefer-default-export */
+import React from 'react';
+import {
+  Alert,
+  Button,
+  Collapsible,
+  OverlayTrigger,
+  Tooltip,
+} from '@openedx/paragon';
+import { FormattedMessage, FormattedNumber, useIntl } from '@edx/frontend-platform/i18n';
+
+import { LoadingSpinner } from '../../generic/Loading';
+import { CodeEditor, EditorAccessor } from '../../generic/CodeEditor';
+import { useLibraryContext } from '../common/context';
+import {
+  useContentLibrary,
+  useUpdateXBlockOLX,
+  useXBlockAssets,
+  useXBlockOLX,
+} from '../data/apiHooks';
+import messages from './messages';
+
+interface Props {
+  usageKey: string;
+}
+
+export const ComponentAdvancedInfo: React.FC<Props> = ({ usageKey }) => {
+  const intl = useIntl();
+  const { libraryId } = useLibraryContext();
+  const { data: library } = useContentLibrary(libraryId);
+  const canEditLibrary = library?.canEditLibrary ?? false;
+  const { data: olx, isLoading: isOLXLoading } = useXBlockOLX(usageKey);
+  const { data: assets, isLoading: areAssetsLoading } = useXBlockAssets(usageKey);
+  const editorRef = React.useRef<EditorAccessor | undefined>(undefined);
+  const [isEditingOLX, setEditingOLX] = React.useState(false);
+  const olxUpdater = useUpdateXBlockOLX(usageKey);
+  const updateOlx = React.useCallback(() => {
+    const newOLX = editorRef.current?.state.doc.toString();
+    if (!newOLX) {
+      /* istanbul ignore next */
+      throw new Error('Unable to get OLX string from codemirror.'); // Shouldn't happen.
+    }
+    olxUpdater.mutateAsync(newOLX).then(() => {
+      // Only if we succeeded:
+      setEditingOLX(false);
+    }).catch(() => {
+      // On error, an <Alert> is shown below. We catch here to avoid the error propagating up.
+    });
+  }, [editorRef, olxUpdater, intl]);
+  return (
+    <Collapsible
+      styling="basic"
+      title={intl.formatMessage(messages.advancedDetailsTitle)}
+    >
+      <dl>
+        <dt><FormattedMessage {...messages.advancedDetailsUsageKey} /></dt>
```

Comment:
Not sure if this matters for the MVP, but we are using a different style across the sidebar
```suggestion
        <h3 className="h5"><FormattedMessage {...messages.advancedDetailsUsageKey} /></h3>
```